### PR TITLE
Fix memory leaks in normal FitPanel

### PR DESCRIPTION
**Pull request by linev**
TF1 objects, created in the TFitEditor, were not destroyed properly.
Also use more clear c++11 syntax for STL containers
Do not generate long names like "PrevFit-7-PrevFit-5-PrevFit-1" for fit functions

### CODE REVIEW

File: gui/fitpanel/src/TFitEditor.cxx [hunk 1]
```diff
@@ -260,6 +255,7 @@ TF1* copyTF1(TF1* f)
          fnew->Save(xmin,xmax,0,0,0,0);
       fnew->SetParent( 0 );
       fnew->AddToGlobalList(false);
+      printf("Create TF1 %s %p\n", fnew->GetName(), fnew);
```
Comment (Axel-Naumann) [inline gui/fitpanel/src/TFitEditor.cxx]: Do we need this line?

File: gui/fitpanel/src/TFitEditor.cxx [hunk 2]
```diff
@@ -2181,11 +2182,10 @@ void TFitEditor::DoFit()
 
    // Save fit data for future use as a PrevFit function.
    TF1* tmpTF1 = copyTF1(fitFunc);
-   ostringstream name;
-   name << "PrevFit-" << fPrevFit.size() + 1;
-   if ( strcmp(tmpTF1->GetName(), "PrevFitTMP") != 0 )
-      name << "-" << tmpTF1->GetName();
-   tmpTF1->SetName(name.str().c_str());
+   TString name = TString::Format("PrevFit-%d", (int) fPrevFit.size() + 1);
+   if (!strstr(fitFunc->GetName(),"PrevFit"))
```
Comment (Axel-Naumann) [inline gui/fitpanel/src/TFitEditor.cxx]: Why did we need to check for `"PrevFitTMP"` before? Was this simply a bug? What does `git blame` say - did we use `"PrevFitTMP"` some time in the past?

File: gui/fitpanel/src/TFitEditor.cxx (hunk 1 above)
Comment (linev) [reply gui/fitpanel/src/TFitEditor.cxx]: No, line was removed with last commit.

File: gui/fitpanel/src/TFitEditor.cxx (hunk 2 above)
Comment (linev) [reply gui/fitpanel/src/TFitEditor.cxx]: "PrevFitTMP" is name of object created right before fit.
But than it was renamed into something like "PrevFit-3". 
If next time such function was used for fitting, result name was "PrevFit-5-PrevFit-3".
And so on.

**Pull request merged**